### PR TITLE
feat: Upgrade Docsy to support Mermaid diagrams

### PR DESCRIPTION
**Pull request by jkuester**
The [new version of Docsy](https://github.com/google/docsy/blob/main/CHANGELOG.md#060) added built-in support for [Mermaid Diagrams](https://mermaid.js.org/). Mermaid is a library that allows for rendering SVG diagram images based on text configuration (that can be easily stored in Markdown files.  (GitHub [added support](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/) for Mermaid diagrams last year.  

The main value that I see from using this kind of diagram is that it makes it easy to update/evolve the diagram in the future. It is difficult to make even minor changes to screenshots of Powerpoint slides, etc without having the original source file used to generate the image. Mermaid diagrams, on the other hand, fit very cleanly into the Markdown workflow and can be created/edited with just a text editor.  

## Changes

- Update Docsy version from `v0.4.0` to `v0.6.0` (latest)
- Update minimum Hugo version to `0.93.0` for Docsy compatibility (CI is already building with `0.101.0`)
- Update `_partial_docker_setup.md` because of [breaking changes](https://github.com/google/docsy/blob/main/CHANGELOG.md#051) to the tabbed panes in `0.5.1`
- Add new diagram for cht-monitoring
- Update several existing diagrams as examples of Mermaid functionality

## Diagrams

For some of the diagrams (the ones without images), I can actually include a preview of the diagram, itself, in this description since GitHub can render them.  My general goal with these diagrams was just to provide an example of the kinds of things we can do with Mermaid. Ultimately, I am pretty happy with how they turned out, but I  have no problems if we decide to revert back to the screenshots for any (or all) of them.  My impressions after working a bit with Mermaid is that it is excellent for creating relatively small diagrams for technical communication, especially when the content may change in the future. It is less suitable for creating professional quality diagrams for use in marketing materials. The layout of the diagrams is generated automatically and it can be challenging to achieve pixel-perfect placement of all the elements. So, Mermaid is probably not a good choice in those scenarios.  

### ~~cht-monitoring UML diagram~~

Mermaid supports many different types of diagrams. The [`flowchat` diagram](https://mermaid.js.org/syntax/flowchart.html) is most useful for creating simple "Powerpoint-style" diagrams where you just connect shapes wit lines.  However, [C4 UML Diagrams](https://github.com/plantuml-stdlib/C4-PlantUML/blob/master/README.md) offer a lot more data-density when diagramming a technical system. I know UML it not really "cool" any more, but I thought I would work up an example diagram with it just to see what folks thought....

<details><summary>New Diagram</summary>

```mermaid
C4Container
Person(admin, "Admin", "Monitors the CHT system")
System_Boundary(cht-monitoring, "CHT Monitoring") {
   Container(grafana, "Grafana", "Web App", "Provides dashboards and alerting")
   ContainerDb(prometheus, "Prometheus", "Time-series DB", "Stores and aggregates metrics")
   Container(json-exporter, "json-exporter", "Prometheus Exporter", "Converts /monitoring JSON to metrics")
   Container(postgres-exporter, "postgres-exporter", "Prometheus Exporter", "Converts Postgres data to metrics")
   Rel(prometheus, json-exporter, "Reads - 5m", "HTTPS")
   UpdateRelStyle(prometheus, json-exporter, $offsetX="5")
   Rel(prometheus, postgres-exporter, "Reads - 5m", "HTTPS")
   Rel(grafana, prometheus, "Reads - 1m", "HTTPS")
   UpdateRelStyle(grafana, prometheus, $offsetX="-30", $offsetY="15")
}
System_Boundary(rdbms, "RDBMS") {
   Container_Ext(couch2pg, "Couch2pg", , "Replicates CouchDB data to Postgres")
   ContainerDb_Ext(postgres, "Postgres", "RDBMS", "Stores replicated CHT Data")
   Rel(couch2pg, couchdb , "Reads", "HTTPS")
   Rel(couch2pg, postgres, "Reads/Writes - 1h", "postgres")
   UpdateRelStyle(couch2pg, postgres, $offsetY="20", $offsetX="-45")
}
System_Boundary(cht, "CHT") {
   Container_Ext(api, "API", , "CHT Backend server")
   ContainerDb_Ext(couchdb, "CouchDB", "NoSQL DB", "Main CHT Datastore")
   Rel(api, couchdb, "Reads/Writes", "HTTPS")
   UpdateRelStyle(api, couchdb, $offsetY="20", $offsetX="-35")
}
Rel(json-exporter, api, "Reads", "HTTPS")
Rel(postgres-exporter, postgres, "Reads", "postgres")
UpdateRelStyle(postgres-exporter, postgres, $offsetY="-30", $offsetX="10")
BiRel(admin, grafana, "Monitors/Alerts", "HTTPS/SMTP/Slack/Etc")
UpdateRelStyle(admin, grafana, $offsetY="-30")
UpdateLayoutConfig($c4ShapeInRow="2", $c4BoundaryInRow="1")
```

</details> 

### Interoperability

This is a super simple diagram.  The notable example here is that [Fontawesome icons](https://fontawesome.com/) are supported in the labels (used them for the arrows)! (Unfortunately, I had to include a screenshot here since GitHub does not support FA in diagram (but Hugo/Docsy does).

<details><summary>Old Diagram</summary>

image

</details> 

<details><summary>New Diagram</summary>

image

</details> 

### ~~Architecture~~

Mermaid flowchart diagrams support some pretty cool options for customizing the style.  I particularly like having the ability to define multiple different style "classes" that can be applied to different elements throughout the diagram.

<details><summary>Old Diagram</summary>

image

</details> 

<details><summary>New Diagram</summary>

```mermaid
flowchart TB
linkStyle default stroke-width:1px
subgraph client
  classDef client_node fill:#dfeaea,stroke:#7cb3b3
  classDef client_device_node fill:#f0f4fd,stroke:#84a1f0
  
  subgraph app_browser[App Browser]
    pouchdb[(PouchDB)]:::client_node
    style app_browser fill:#dfeaea,stroke:#7cb3b3
  end
  
  subgraph app_android[Android App]
    app_android_browser[Browser]:::client_node
    style app_android fill:#dfeaea,stroke:#7cb3b3
  end
  
  integrations["`Integrations
    with other
    systems`"]:::client_node
  browser[Browser]:::client_node
  sms_aggregator["`SMS 
    aggregator`"]:::client_node
  sms_gateway["`SMS 
    gateway`"]:::client_node
  feature_phones[Feature phones]:::client_device_node
  smartphones_tables_computers["`Smartphones, tablets, 
    and computers`"]:::client_device_node
  feature_phones <--> sms_gateway & sms_aggregator
  smartphones_tables_computers <--> app_browser & app_android
end
subgraph server
  subgraph cht_core[CHT Core Framework]
    classDef cht_core_node fill:#eef5f9,stroke:#68a5c8
    couchdb[(CouchDB)]:::cht_core_node
    haproxy(HAProxy):::cht_core_node
    upgrade_service("`Upgrade 
      Service`"):::cht_core_node
    api(API):::cht_core_node
    sentinel(Sentinel):::cht_core_node
    nginx(NGINX):::cht_core_node
    haproxy --> couchdb
    api --> upgrade_service & haproxy
    sentinel --> api
    nginx --> api
    style cht_core fill:none,stroke:#68a5c8,stroke-width:1px,color:#68a5c8
  end
  subgraph cht_sync[CHT Sync]
    classDef cht_sync_node fill:#fcf6e7,stroke:#ebb338
    postgres[(PostgreSQL)]:::cht_sync_node
    postgrest(PostgREST):::cht_sync_node
    dbt(DBT):::cht_sync_node
    logstash(Logstash):::cht_sync_node
    superset(Superset):::cht_sync_node
    postgrest --> postgres
    dbt --> postgres
    logstash --> postgrest
    superset ----> postgres    
    style cht_sync fill:none,stroke:#ebb338,stroke-width:1px,color:#ebb338
  end
  couchdb --> logstash
end
browser <--> superset
integrations & pouchdb & app_android_browser & sms_aggregator & sms_gateway <--> nginx
classDef outer_group fill:none,stroke-dasharray: 5 5
class client,server outer_group
```

</details> 

### Hierarchy

It is also possible to inject custom images into Mermaid flowchart diagrams. Of course, due to browser security stuff, you cannot really just hot-load the images from just anywhere, but you can load local page resources in Hugo (similar to how we currently supply screenshot images, etc).

#### Places

<details><summary>Old Diagram</summary>

image

</details> 

<details><summary>New Diagram</summary>

image

</details> 

### People

<details><summary>Old Diagram</summary>

image

</details> 

<details><summary>New Diagram</summary>

image

</details>

### CODE REVIEW

File: content/en/apps/guides/hosting/monitoring/setup.md [hunk 1]
```diff
@@ -29,6 +29,48 @@ The solution provides both an overview dashboard as well as a detail dashboard.
 
 ### Setup
 
+```mermaid
```
Comment (mrjones-plip) [inline content/en/apps/guides/hosting/monitoring/setup.md:32]: this diagram is complex and also in need of a number of fixes.  For example, this area is illegible because of overlapping text and lines: 

image

let's break adding this diagram out into it's own PR so as to not block hugo uplift and adding mermaid feature

File: content/en/apps/tutorials/_partial_docker_setup.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ title: "Local Docker Setup"
 toc_hide: true
 hide_summary: true
 ---
-{{< tabpane code=false >}}
+{{< tabpane text=true >}}
```
Comment (mrjones-plip) [inline content/en/apps/tutorials/_partial_docker_setup.md:6]: nice job catching the breaking change and adding a fix!

File: content/en/core/overview/architecture.md [hunk 3]
```diff
@@ -8,7 +8,80 @@ description: >
 
 ## Overview
 
-![Architecture of a CHT project](architecture.png)
+```mermaid
```
Comment (mrjones-plip) [inline content/en/core/overview/architecture.md:11]: I think the replacement diagram is aesthetically unappealing and we break this out into it's own PR so we can involve UX/design folks before pushing.  Hopefully we'll get there, but having the odd `.png` where we need pixel perfect art doesn't negate the ease of a quick mermaid diagram!

File: content/en/apps/concepts/hierarchy.md [hunk 4]
```diff
@@ -24,7 +24,28 @@ The information hierarchy is configured in the administration console. The hiera
 
 This is an example of a simple hierarchy that includes a CHW Supervisor area, CHW area, and families as levels which serve as “places” or units of organizing people. 
 
-![hierarchy-with-places.png](hierarchy-with-places.png)
+```mermaid
```
Comment (mrjones-plip) [inline content/en/apps/concepts/hierarchy.md:27]: OMG!! LOVE LOVE LOVE This use of mermaid! Perfect.

**Review by mrjones-plip — CHANGES_REQUESTED** [review]
@jkuester  - amazing work here! The benefits of, for example, being able to do native hierarchy changes in your editor that include custom icons is bananas-amazing-cool.

that said, two of the diagrams are far enough away from the existing ones ("Architecture of CHT Instances") or have some fatal flaws ("monitoring Setup") we should break them into their own PRs so we can discretely work on them w/o blocking this PR>

File: content/en/apps/guides/hosting/monitoring/setup.md (hunk 1 above)
Comment (jkuester) [reply content/en/apps/guides/hosting/monitoring/setup.md]: Good call. I will open a separate PR for this.

**Review by mrjones-plip — APPROVED** [review]
🍝

**Review by esthermmoturi — APPROVED** [review]
LGTM!
